src/styles/main.scss

```diff
@@ -0,0 +1,35 @@
+.main {
+  justify-content: center;
+  margin: 0;
+  padding: 50px 40px;
+  display: grid;
+
+  --column-count: 4;
+
+  grid-template-columns: repeat(var(--column-count), 200px);
```

Comment:
The grid-template-columns property is correctly set to use the '--column-count' variable, ensuring the layout adapts to different screen sizes as specified in the task.